nominal/core/containerized_extractors.py

```diff
@@ -0,0 +1,281 @@
+            auth = UserPassAuth._from_conjure(raw_source.authentication.user_and_password)
+        else:
+            raise ValueError(f"Unexpected type for authentication: {raw_source.authentication.type}")
+
+        return cls(
+            registry=raw_source.registry,
+            repository=raw_source.repository,
+            tag_details=TagDetails._from_conjure(raw_source.tag_details),
+            authentication=auth,
+            command=raw_source.command,
+        )
+
+    def _to_conjure(self) -> ingest_api.DockerImageSource:
+        if self.authentication is None:
+            authentication = ingest_api.Authentication(public=ingest_api.PublicAuthentication())
+        else:
+            authentication = self.authentication._to_conjure()
+
+        return ingest_api.DockerImageSource(
+            authentication=authentication,
+            registry=self.registry,
+            repository=self.repository,
+            tag_details=self.tag_details._to_conjure(),
+        )
+
+
+class FileOutputFormat(Enum):
+    PARQUET_TAR = "PARQUET_TAR"
+    CSV = "CSV"
+    PARQUET = "PARQUET"
+
+    @classmethod
+    def _from_conjure(cls, raw_source: ingest_api.FileOutputFormat) -> Self:
+        return cls(raw_source.value)
+
+    def _to_conjure(self) -> ingest_api.FileOutputFormat:
+        return ingest_api.FileOutputFormat(self.value)
+
+
+@dataclass(frozen=True)
+class TimestampMetadata:
+    """Metadata about the timestamp output from the extractor.
+
+    Args:
+        series_name: Name of the column containing timestamp data in the output files.
+        timestamp_type: Type information specifying how timestamps should be interpreted.
+    """
+
+    series_name: str
+
+    timestamp_type: _ConjureTimestampType
+
+    @classmethod
+    def _from_conjure(cls, raw_metadata: ingest_api.TimestampMetadata) -> Self:
+        return cls(
+            series_name=raw_metadata.series_name,
+            timestamp_type=_ConjureTimestampType._from_conjure(raw_metadata.timestamp_type),
+        )
+
+    def _to_conjure(self) -> ingest_api.TimestampMetadata:
+        return ingest_api.TimestampMetadata(
+            series_name=self.series_name,
+            timestamp_type=self.timestamp_type._to_conjure_ingest_api(),
+        )
+
+
+@dataclass(frozen=True)
+class ContainerizedExtractor(HasRid):
+    """Containerized extractor which can be used to parse custom data formats into Nominal using docker images."""
+
+    rid: str
+
+    name: str
+
+    description: str | None
+
+    image: DockerImageSource
+
+    inputs: Sequence[FileExtractionInput]
+
+    properties: Mapping[str, str]
+
+    labels: Sequence[str]
+
+    timestamp_metadata: TimestampMetadata
+
+    _clients: _Clients = field(repr=False)
+
+    class _Clients(HasScoutParams, Protocol):
+        @property
+        def containerized_extractors(self) -> ingest_api.ContainerizedExtractorService: ...
+
+    def update(
+        self,
+        name: str | None = None,
```

Comment:
```suggestion
        self,
        *,
        name: str | None = None,
```

I think we similarly do kw-only for update APIs